tabs/utils/service-control.sh

```diff
@@ -0,0 +1,271 @@
+        echo "Failed to enable service: $SERVICE_NAME."
+    fi
+}
+
+# Function to create, start, and enable a service from an external service file
+create_service_from_external() {
+
+    # List all .service files in the SCRIPT_DIR
+    echo "============================"
+    echo "Listing available service files"
+    echo "============================"
+    for FILE in "$SCRIPT_DIR"/*.service; do
+        echo "$(basename "$FILE")"
+    done
+
+    echo "Enter the filename (without the .service extension) of the service to create:"
+    read -r SERVICE_NAME
+
+    SERVICE_FILE="$SCRIPT_DIR/$SERVICE_NAME.service"
+
+    if [ ! -f "$SERVICE_FILE" ]; then
+        echo "Service file $SERVICE_FILE does not exist."
+        return
+    fi
+
+    echo "Enter the username to run the service as (leave empty for no specific user):"
+    read -r SERVICE_USER
+
+    # Create the systemd service file path
+    SYSTEMD_SERVICE_FILE="/etc/systemd/system/$SERVICE_NAME.service"
+
+    # Add or update the User= line in the service file
+    if [ -n "$SERVICE_USER" ]; then
+        # Check if User= exists and append username if needed
+        if grep -q '^User=' "$SERVICE_FILE"; then
+            # Update the existing User= line with the new username
+            sed -i "s/^User=.*/User=$SERVICE_USER/" "$SERVICE_FILE"
+        else
+            # Add the User= line if it doesn't exist
+            sed -i '/^\[Service\]/a User='"$SERVICE_USER" "$SERVICE_FILE"
+        fi
+    fi
+
+    # Copy the modified service file to /etc/systemd/system/
+    $ESCALATION_TOOL cp "$SERVICE_FILE" "$SYSTEMD_SERVICE_FILE"
+
+    # Set permissions and reload systemd
+    $ESCALATION_TOOL chmod 644 "$SYSTEMD_SERVICE_FILE"
+    $ESCALATION_TOOL systemctl daemon-reload
+    echo "Service $SERVICE_NAME has been created and is ready to be started."
+
+    # Optionally, enable and start the service
+    echo "Do you want to start and enable the service now? (y/n)"
+    read -r START_ENABLE
+
+    if [ "$START_ENABLE" = "y" ]; then
+        $ESCALATION_TOOL systemctl start "$SERVICE_NAME"
+        $ESCALATION_TOOL systemctl enable "$SERVICE_NAME"
+        echo "Service $SERVICE_NAME has been started and enabled."
+    else
+        echo "Service $SERVICE_NAME has been created but not started."
+    fi
+}
+
+main() {
+    while true; do
+        show_menu
+        echo "Enter your choice:"
+        read -r CHOICE
+
+        case $CHOICE in
+            1) view_all_services ;;
+            2) view_enabled_services ;;
+            3) view_disabled_services ;;
+            4) add_service ;;
+            5) remove_service ;;
+            6) start_service ;;
+            7) stop_service ;;
+            8) enable_service ;;
+            9) disable_service ;; 
+            10) create_service_from_external ;;
+            11) echo "Exiting..."; exit 0 ;;
+            *) echo "Invalid choice. Please try again." ;;
+        esac
+
+        echo "Press [Enter] to continue..."
+        read -r
```

Comment:
```suggestion
        read
```

1. this is not posix compliant
2. i don't think `-r` is really necessary

Reply:
https://github.com/ChrisTitusTech/linutil/pull/419#discussion_r1761516772